test/integration/thruster.cc

```diff
@@ -285,14 +314,50 @@ void ThrusterTest::TestWorld(const std::string &_world,
       {
         EXPECT_NEAR(0.0, angVel.X(), _baseTol);
       }
+      else if (_namespace == "deadband")
+      {
+        continue;
+      }
       else
       {
-        EXPECT_NEAR(omega, angVel.X(), omegaTol) << i;
+        ASSERT_NEAR(omega, angVel.X(), omegaTol) << i;
       }
     }
     EXPECT_NEAR(0.0, angVel.Y(), _baseTol);
     EXPECT_NEAR(0.0, angVel.Z(), _baseTol);
   }
+
+  modelPoses.clear();
+  propellerAngVels.clear();
+  propellerLinVels.clear();
+  // Make sure that when the deadband is disabled
+  // commands below the deadband should create a movement
+  auto latest_pose = modelPoses.back();
+  msgs::Boolean db_msg;
+  if (_namespace == "deadband")
+  {
+    // disable the deadband
+    db_msg.set_data(false);
+    db_pub.Publish(db_msg);
+    // And we send a command that is still below the deadband threshold
+    msg.set_data(_deadband / 2.0);
+    pub.Publish(msg);
+    // Check movements
+    fixture.Server()->Run(true, 1000, false);
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
To the best of my knowledge, there is no need for this sleep. You already run the server in blocking mode.

Reply:
I simply copied this block from a few lines above: https://github.com/gazebosim/gz-sim/blob/4eeed347a12e3e63f61b7f0148ce41b947d8ec98/test/integration/thruster.cc#L207-L208

I also don't know why it was necessary there. I'll try re-running the test without the sleep and tell you if it works.

Reply:
Note that the sleep is also present here:
https://github.com/gazebosim/gz-sim/blob/42c285ef1a7b35d7e64c8fcdc4f270fb2b42b9e3/test/integration/thruster.cc#L216-L217

Reply:
Thanks for pointing this out.

Reply:
To clarify, the sleep is also probably not needed in the above test cases you've pointed out. I believe its best new code does not use the sleep.